src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -581,6 +644,19 @@ def check_pro_client_prerequisites(self):
         self.composite_logger.log_debug("Ubuntu Pro Client pre-requisite checks:[IsFeatureEnabled={0}][IsOSVersionCompatible={1}][IsPythonCompatible={2}][Error={3}]".format(Constants.UbuntuProClientSettings.FEATURE_ENABLED, self.__get_os_major_version() <= Constants.UbuntuProClientSettings.MAX_OS_MAJOR_VERSION_SUPPORTED, self.__is_minimum_required_python_installed(), exception_error))
         return self.__pro_client_prereq_met
 
+    def set_security_esm_package_status(self, operation):
+        """Set the security-ESM classification for the esm packages."""
+        security_esm_update_query_success, security_esm_updates, security_esm_updates_versions = self.get_security_esm_updates()
+        if self.__pro_client_prereq_met and security_esm_update_query_success and len(security_esm_updates) > 0:
+            self.telemetry_writer.write_event("set Security-ESM package status:[Operation={0}][Updates={1}]".format(operation, str(security_esm_updates)), Constants.TelemetryEventLevel.Verbose)
+            if operation == Constants.ASSESSMENT:
+                self.status_handler.set_package_assessment_status(security_esm_updates, security_esm_updates_versions, Constants.PackageClassification.SECURITY_ESM)
+                # If the Ubuntu Pro Client is not attached, set the error with the code UA_ESM_REQUIRED. This will be used in portal to mark the VM as unattached to pro.
+                if not self.ubuntu_pro_client.ubuntu_pro_client_is_attached:
+                    self.status_handler.add_error_to_status("{0} patches requires Ubuntu Pro for Infrastructure with Extended Security Maintenance".format(len(security_esm_updates)), Constants.PatchOperationErrorCodes.UA_ESM_REQUIRED)
```

Comment:
Is this error not needed for Installation operation?

Reply:
We do this in L700. I do this because, we will get a count of patches that are part of patching operation and cannot be patched. 
In this place, we will be getting count of all availalbe esm packages.